tests/unit/auth.tcl

```diff
@@ -25,3 +25,43 @@ start_server {tags {"auth"} overrides {requirepass foobar}} {
         r incr foo
     } {101}
 }
+
+start_server {tags {"auth_binary_password"}} {
+    test {AUTH fails when binary password is wrong} {
+        r config set requirepass "abc\x00def"
+        catch {r auth abc} err
+        set _ $err
+    } {WRONGPASS*}
+
+    test {AUTH succeeds when binary password is correct} {
+        r config set requirepass "abc\x00def"
+        r auth "abc\x00def"
+    } {OK}
+
+    start_server {tags {"masterauth"}} {
+        set master [srv -1 client]
+        set master_host [srv -1 host]
+        set master_port [srv -1 port]
+        set slave [srv 0 client]
+
+        test {MASTERAUTH test with binary password} {
+            $master config set requirepass "abc\x00def"
+
+            # Configure the replica with masterauth
+            $slave slaveof $master_host $master_port
+            $slave config set masterauth "abc"
+
+            # sleep for 3 seconds and verify replica is not in sync with master
+            $slave debug sleep 3
```

Comment:
two things i don't like about this line:
1. it uses DEBUG SLEEP which we better avoid unless we want to put the server in a sleep and prevent it from talking to other clients. i rather use `after 3000` in tests where possible.
2. i rather not have this test wait for 3 seconds unnecessarily (makes the test suite longer). in any place we can, we should use a wait_for_ loop that exists as soon as possible rather than wait excessively.  in this case a `wait_for_log_messages` on "Unable to AUTH to MASTER" would probably be good. (other alternatives are ACL LOG, and maybe some INFO STATS field).

Reply:
Fixed